components/gateway/internal/gqlschema/schema.graphql

```diff
@@ -89,19 +91,26 @@ type Version {
     deprecatedSince: String
     """if true, will be removed in the next version"""
     forRemoval: Boolean
-
 }
 
 type APIDefinition {
     id: ID!
     spec: APISpec
     targetURL: String!
-    credential: Credential
-    additionalHeaders: HttpHeaders #
-    additionalQueryParams: QueryParams
+    """ group allows you to find the same API but in different version """
+    group: String
+    """"If runtime does not exist, an error will be returned. If runtime exists but does not have Auth defined, null is returned"""
+    auth(runtimeID: ID!): Auth
```

Comment:
I think having an additional default `Auth` for each `APIDefinition` would be preferable. In that case, `runtimeID` would be optional, and when no `runtimeID` is specified or runtime with specified `runtimeID` wasn't configured with specific `Auth` the default `Auth` would be returned.

Reply:
I added separateField defaultAuth, let me know what you think!

Reply:
I like this approach as well, very nice :) 👍